src/environment/stella_environment.cpp

```diff
@@ -78,13 +79,15 @@ void StellaEnvironment::reset() {
   emulate(RESET, PLAYER_B_NOOP, m_num_reset_steps);
 
   // reset the rom (after emulating, in case the NOOPs led to reward)
-  m_settings->reset();
+  m_settings->reset(m_osystem->console().system(), *(this));
```

Comment:
From a design perspective it would be good to avoid a cyclical dependency between Environment and RomSettings. Was this done because each game needs custom code to switch modes?